### PR TITLE
KEYCLOAK-9008 CookieStoreRootContextTest stabilization

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/KEYCLOAK-9008

### CODE REVIEW

**Comment by slaskawi** [comment]
Started some more builds: 

* https://keycloak-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/keycloak-pr-pipeline-structured/1749/
* https://keycloak-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/keycloak-pr-pipeline-structured/1750/
* https://keycloak-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/keycloak-pr-pipeline-structured/1751/
* https://keycloak-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/keycloak-pr-pipeline-structured/1752/

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java [hunk 1]
```diff
@@ -54,15 +53,6 @@
 @AppServerContainer(ContainerConstants.APP_SERVER_EAP6)
 public class CookieStoreRootContextTest extends DemoServletsAdapterTest {
```
Comment (vramik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: imho `CookieStoreRootContextTest` shouldn't inherit from `DemoServletsAdapterTest` but rather from `AbstractServletsAdapterTest` 

In this state it runs all tests from `DemoServletsAdapterTest` twice.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java [hunk 2]
```diff
@@ -54,15 +53,6 @@
...
 
-    // Javascript browser needed KEYCLOAK-4703
-    @Drone
-    @JavascriptBrowser
-    protected WebDriver jsDriver;
-
-    @Page
-    @JavascriptBrowser
-    protected OIDCLogin jsDriverTestRealmLoginPage;
-
```
Comment (vramik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: I'm not sure about this removal, are we sure the test works with htmlunit?

**Review by vramik — CHANGES_REQUESTED** [review]
@slaskawi thank you for the PR, I have just few comments

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java (hunk 2 above)
Comment (slaskawi) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: Both Travis and Jenkins are green. I'm not sure how to perform additional checks...

Seems fine...

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java (hunk 1 above)
Comment (slaskawi) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: That's true. However, I think it's intentional (or partially intentional).

The `testTokenInCookieSSORoot` uses `customerPortal` variable, which is defined in `DemoServletsAdapterTest`. Also, `loginToCustomerCookiePortalRoot` uses `assertLogged`, which again, is defined in `DemoServletsAdapterTest`. 

So, I guess we can either leave this like this or move `customer-cookie-portal-root` deployment into `DemoServletsAdapterTest` and simply remove `CookieStoreRootContextTest`. 

WDYT?

Comment (vramik) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: The latter option seems better to me. I'd go with this one.

Comment (slaskawi) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: Ok, I'll try it out tomorrow and push an update. Thanks for the feedback.

Comment (slaskawi) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: Implemented

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java (hunk 2 above)
Comment (slaskawi) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/CookieStoreRootContextTest.java]: I'm resolving this discussion. Please reopen it if you have any more concerns or ideas how to test it.

**Comment by vramik** [comment]
@slaskawi thanks a lot for the update, lgtm. There is DemoFilterServletAdapterTest>DemoServletsAdapterTest.testTokenInCookieSSORoot failing in travis. I guess the test isn't supposed to work with filter adapter, is it?

**Comment by slaskawi** [comment]
@vramik I don't think so. The test looks fine and running it alone ends with a success. There must be something else that breaks it.

**Comment by slaskawi** [comment]
@vramik Ohhh crap, I was looking on a wrong file! Silly me, the fix is on its way...

**Comment by vramik** [comment]
@slaskawi LGTM, waiting for ci tests to pass. It seems there is an issue: KEYCLOAK-9121

**Review by vramik — APPROVED** [review]

**Review by pdrozd — APPROVED** [review]
Based on vramik's approval

**Pull request merged**